identity-account/src/account/account.rs

```diff
@@ -87,10 +86,18 @@ impl Account {
 
   /// Creates a new identity and returns an [`Account`] instance to manage it.
   /// The identity is stored locally in the [`Storage`] given in [`AccountSetup`], and published
-  /// using the [`ClientMap`].
+  /// using the [`Client`].
   ///
   /// See [`IdentitySetup`] to customize the identity creation.
   pub(crate) async fn create_identity(setup: AccountSetup, input: IdentitySetup) -> Result<Self> {
+    // Error if the DID network does not match the client on the Account.
+    // TODO: automatically use client network?
```

Comment:
I thought about it (hence the TODO) and I'm in favour of removing it. The only problem I can see is that the network used depends entirely on the client, which could cause subtle bugs/confusion since the network parameter on `Client`/`ClientBuilder` defaults to the `Mainnet`. If I recall correctly there either isn't an error message when publishing a document to the wrong network or the error is vague. 

I'll make the change with added safeguards in `Client` when publishing and hopefully it won't cause any issues.